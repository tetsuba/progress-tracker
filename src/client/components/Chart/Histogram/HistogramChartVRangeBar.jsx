import React from 'react'

type Props = {
  correct?: boolean,
  wrong?: boolean,
  height: number,
}

export default function HistogramChartVRangeBar(props: Props) {
  const { correct, height } = props
  const style = {
    width: '100%',
    height: `${height}%`,
    backgroundColor: correct ? '#309c3f' : 'red',
    boxSizing: 'border-box',
    borderTop: 'solid 1px #CCC',
  }

  return <div style={style} />
}
