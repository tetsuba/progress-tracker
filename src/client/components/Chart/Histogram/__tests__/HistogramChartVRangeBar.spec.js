import { testRenderer } from '../../../../../test/testHelper'
import HistogramChartVRangeBar from '../HistogramChartVRangeBar'

describe('<HistogramChartVRangeBar>', () => {
  describe('@Render', () => {
    it('Should render a true value', () => {
      const props = {
        correct: true,
        wrong: false,
        height: 20,
      }
      const wrapper = testRenderer(HistogramChartVRangeBar, props)
      expect(wrapper).toMatchSnapshot()
    })

    it('Should render a false value', () => {
      const props = {
        correct: true,
        wrong: false,
        height: 20,
      }
      const wrapper = testRenderer(HistogramChartVRangeBar, props)
      expect(wrapper).toMatchSnapshot()
    })
  })
})
