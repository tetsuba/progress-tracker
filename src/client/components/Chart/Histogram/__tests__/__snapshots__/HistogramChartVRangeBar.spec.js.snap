// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistogramChartVRangeBar> @Render Should render a false value 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#309c3f",
      "borderTop": "solid 1px #CCC",
      "boxSizing": "border-box",
      "height": "20%",
      "width": "100%",
    }
  }
/>
`;

exports[`<HistogramChartVRangeBar> @Render Should render a true value 1`] = `
<div
  style={
    Object {
      "backgroundColor": "#309c3f",
      "borderTop": "solid 1px #CCC",
      "boxSizing": "border-box",
      "height": "20%",
      "width": "100%",
    }
  }
/>
`;
